Dynamic_Memory_Allocation_For_User_Input/buffer.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#define INITIAL_SIZE 2000U
+#define MAX_UINT32_TO_STR_BASE10_COUNT 10U
+
+typedef struct
+{
+    uint32_t inputBufferSize;
+    uint32_t inputBufferCapacity;
+    char* p_inputBuffer;
```

Comment:
Make simpler names, you already know it is an input buffer by the name of the structure.